public/js/chat.js

```diff
@@ -80,6 +73,17 @@ const autoscroll = () => {
     }
 }
 
+const { username, room, roomId } = Qs.parse(location.search, {
+    ignoreQueryPrefix: true,
+})
+
+const socket = io({
+    query: {
```

Comment:
How does the data get here?

Reply:
Which data? query ? it is the same as before - from url

Reply:
But the old code didn't pass anything to `io`. What does the passing mean?

Reply:
But the old code didn't pass anything to `io`. What does the passing mean?